backend/app/readers/reader/reader.py

```diff
@@ -0,0 +1,12 @@
+from abc import ABC, abstractmethod
+from collections.abc import Generator
+
+from app.models.record import Record
+
+
+class Reader(ABC):
+    """An interface to read and parse Records from storage."""
+
+    @abstractmethod
+    def read(self) -> Generator[Record, None, None]:
```

Comment:
Nit: I'd typically use the more general `Iterable[Record]` rather than `Generator` so that a sequence can be returned OR a generator can be used.

Reply:
Yes, that makes sense. I have changed all `Generators `to `Iterable[Record]`.

Reply:
Except the `lifespan event` in main which is decorated with `@asynccontextmanager`, so has a return type of `AsyncIterator[None]`.